core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3204,6 +3204,14 @@ private[spark] object Utils extends Logging {
     }
     files.toSeq
   }
+
+  val isG1GarbageCollector: Boolean = {
+    ManagementFactory.getGarbageCollectorMXBeans
```

Comment:
When I looked for how to find out what the garbage collector type is I bumped into this several times:

```java
  HotSpotDiagnosticMXBean diagnostic = ManagementFactoryHelper.getDiagnosticMXBean();

    VMOption option = diagnostic.getVMOption("UseG1GC");
    if (option.getValue().equals("false")) {
      ...
    }
```
For example at the [OpenJDK tests](https://www.programcreek.com/java-api-examples/?code=AdoptOpenJDK%2Fopenjdk-jdk8u%2Fopenjdk-jdk8u-master%2Fhotspot%2Ftest%2Fgc%2Farguments%2FTestG1HeapRegionSize.java
). 

Is there any reason why a different solution have been chosen here?

Reply:
emm, seems your solution is more elegant. I'll update it

Reply:
Thanks!